triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceTest.scala

```diff
@@ -616,6 +642,157 @@ trait AbstractTriggerServiceTest extends AbstractTriggerServiceTestHelper {
+        choice = "Archive",
+        choiceArgument = Some(Value().withRecord(Record())),
+      )
+    )
+  }
+
+  val catsAppId: ApplicationId = ApiTypes.ApplicationId("cats-app-id")
+
+  it should "stop the trigger if the ACS is too large at initialization time" inClaims withTriggerService(
+    List(dar),
+    triggerRunnerConfig = Some(
+      DefaultTriggerRunnerConfig
+        .copy(
+          // As the trigger starts with the ACS pre-populated with 100 Cat contracts, we should overflow at startup using 10
+          maximumActiveContracts = 10
+        )
+    ),
+  ) { uri: Uri =>
+    for {
+      client <- sandboxClient(
+        catsAppId,
+        actAs = List(ApiTypes.Party(alice.unwrap)),
+      )
+      adminClient <- sandboxClient(catsAppId, admin = true)
+      _ <- adminClient.partyManagementClient.allocateParty(Some(alice.unwrap), None)
+      // Ensure there are no Cat contracts
+      _ <- getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+        .map(_ shouldBe Vector())
+      // Create 100 Cat contracts
+      _ <- Future.sequence(
+        (1 to 100).map(id => submitCmd(client, alice.unwrap, breedCat(id.toLong)))
+      )
+      // Wait for our Cat contracts to be created
+      _ <- RetryStrategy.constant(20, 1.seconds) { (_, _) =>
+        getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+          .map(_.length shouldBe 100)
+      }
+      // Now allow the trigger to start running
+      resp <- startTrigger(uri, s"$testPkgId:Cats:breedingTrigger", alice, Some(catsAppId))
+      catsTrigger <- parseTriggerId(resp)
+      _ <- assertTriggerIds(uri, alice, Vector(catsTrigger))
+      _ <- assertTriggerStatus(catsTrigger, _ should contain("stopped: runtime failure"))
+      _ <- assertTriggerRunnerStatus(
+        catsTrigger,
+        _ should contain(
+          "[ERROR] Due to an excessive number of active contracts, stopping the trigger"
+        ),
+      )
+    } yield succeed
+  }
+
+  it should "stop the trigger if the ACS overflows at runtime" inClaims withTriggerService(
+    List(dar),
+    triggerRunnerConfig = Some(
+      DefaultTriggerRunnerConfig
+        .copy(
+          // As the trigger creates 100 Cat contracts, we should eventually overflow using 1
+          maximumActiveContracts = 1
+        )
+    ),
+  ) { uri: Uri =>
+    for {
+      client <- sandboxClient(
+        catsAppId,
+        actAs = List(ApiTypes.Party(alice.unwrap)),
+      )
+      _ <- getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+        .flatMap(events =>
+          Future.sequence(events.map { event =>
+            submitCmd(client, alice.unwrap, killCat(event.contractId))
+          })
+        )
+      // Wait until there are no Cat contracts
+      _ <- RetryStrategy.constant(20, 1.seconds) { (_, _) =>
+        getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+          .map(_ shouldBe Vector())
+      }
```

Comment:
Pollution from prior testing requires us to cleanup prior to allowing this test to proceed